Lib/ufo2ft/featureWriters/cursFeatureWriter.py

```diff
@@ -0,0 +1,129 @@
+from fontTools.misc.fixedTools import otRound
+
+from ufo2ft.featureWriters import BaseFeatureWriter, ast
+from ufo2ft.featureWriters.kernFeatureWriter import unicodeScriptDirection
+from ufo2ft.util import classifyGlyphs
+
+
+class CursFeatureWriter(BaseFeatureWriter):
+    """Generate a curs feature base on glyph anchors.
+
+    The default mode is 'skip': i.e. if the 'curs' feature is already present in
+    the feature file, it is not generated again.
+
+    The optional 'append' mode will add extra lookups to an already existing
+    features, if any.
+
+    By default, anchors names 'entry' and 'exit' will be used to connect the
+    'entry' anchor of a glyph with the 'exit' anchor of the preceding glyph.
+    """
+
+    tableTag = "GPOS"
+    features = frozenset(["curs"])
+
+    entryAnchorNames = {"entry"}
+    exitAnchorNames = {"exit"}
+
+    def _makeCursiveFeature(self):
+        cmap = self.makeUnicodeToGlyphNameMapping()
+        if any(unicodeScriptDirection(uv) == "LTR" for uv in cmap):
+            gsub = self.compileGSUB()
+            dirGlyphs = classifyGlyphs(unicodeScriptDirection, cmap, gsub)
+            shouldSplit = "LTR" in dirGlyphs
+        else:
+            shouldSplit = False
```

Comment:
So you only make split lookups if there's any glyph associated with a LTR script whether or not it uses cursive attachment anchors (which is very common, I'd say), otherwise you only make a single lookup (persumably RTL). I guess the assumption is curs is mostly for RTL fonts?

I'd have expected that the split would only occur if you find that, among the glyphs that do have cursive anchors, there are _both_ glyphs associated with LTR and RTL scripts. Not just if there's some LTR glyphs in the cmap

Reply:
Is this related to the logic split for the RTL flag detailed in https://docs.microsoft.com/en-us/typography/opentype/spec/gpos#lookup-type-3-cursive-attachment-positioning-subtable?

Reply:
For `curs` feature whether the `RightToLeft` flag is to be used or not should not be decided based on text direction but on what side of the cursive attachment should set on the baseline. But since there is no way to oath this information in UFO, I assuming that if the font has any RTL glyphs then `RightToLeft` flag should be used is as good guess as any. Or we can ask @schriftgestalt what Glyphs does and try to match it.